relay-server/src/services/store.rs

```diff
@@ -989,21 +994,86 @@ impl StoreService {
             }
         };
 
-        for mut log in logs.items {
-            log.organization_id = scoping.organization_id.value();
-            log.project_id = scoping.project_id.value();
-            log.retention_days = retention_days;
-            log.received = safe_timestamp(received_at);
+        for log in logs.items {
+            let timestamp_seconds = log.timestamp_nanos / 1_000_000_000;
+            let timestamp_nanos = log.timestamp_nanos % 1_000_000_000;
+            let item_id = u128::from_be_bytes(
+                *Uuid::new_v7(uuid::Timestamp::from_unix(
+                    uuid::NoContext,
+                    timestamp_seconds,
+                    timestamp_nanos as u32,
+                ))
+                .as_bytes(),
+            )
+            .to_le_bytes()
+            .to_vec();
+            let received_nanos = if let Some(received) = received_at.timestamp_nanos_opt() {
+                received % 1_000_000_000
+            } else {
+                0
+            };
+            let mut trace_item = TraceItem {
+                item_type: TraceItemType::Log.into(),
+                organization_id: scoping.organization_id.value(),
+                project_id: scoping.project_id.value(),
+                received: Some(Timestamp {
+                    seconds: safe_timestamp(received_at) as i64,
+                    nanos: received_nanos as i32,
+                }),
+                retention_days: retention_days.into(),
+                timestamp: Some(Timestamp {
+                    seconds: timestamp_seconds as i64,
+                    nanos: timestamp_nanos as i32,
+                }),
```

Comment:
You're dropping the `nanos` in the eap item consumer, no? DateTime on eap table only stores 1 second resolution.  We want sub 1-second resolution in logs, presumably we don't want it for all trace items, but for logs can we re-add `timestamp_precise` here in attributes? Or are we fine with keeping `item_id` as uuidv7, which gives ~50ns resolution (good enough)

Reply:
Indeed, we will only store the timestamp with second precision. This is more to generate a proper timestamp, I could remove it, it won't change a thing.

I'll add back `timestamp_precise`.